Data-Structures/Tree/SegmentTree.js

```diff
@@ -14,53 +14,80 @@ class SegmentTree {
   size
   tree
   constructor (arr) {
+    // we define tree like this
+    // tree[1] : root node of tree
+    // tree[i] : i'th node
+    // tree[i * 2] : i'th left child
+    // tree[i * 2 + 1] : i'th right child
+    // and we use bit, shift operation for index
     this.size = arr.length
-    this.tree = new Array(2 * this.size)
+    this.tree = new Array(2 * arr.length)
     this.tree.fill(0)
 
     this.build(arr)
   }
 
+  // function to build the tree
   build (arr) {
-    const n = this.size
+    const { size, tree } = this
     // insert leaf nodes in tree
-    for (let i = 0; i < n; i++) {
-      this.tree[n + i] = arr[i]
+    // leaf nodes will start from index N
+    // in this array and will go up to index (2 * N – 1)
+    for (let i = 0; i < size; i++) {
+      tree[size + i] = arr[i]
     }
 
     // build the tree by calculating parents
-    for (let i = n - 1; i > 0; --i) {
-      this.tree[i] = this.tree[i << 1] + this.tree[(i << 1) | 1]
+    // tree's root node will contain all leaf node's sum
+    for (let i = size - 1; i > 0; --i) {
+      // current node's value is the sum of left child, right child
+      // tree[i] = tree[i * 2] + tree[i * 2 + 1]
```

Comment:
this comment is quite literally redundant :sweat_smile:

Reply:
😅 OMG! I forgot to erase that line. thanks for your review